pkg/workspace/image/puller.go

```diff
@@ -0,0 +1,66 @@
+package image
+
+import (
+	"bytes"
+	"text/template"
+
+	_ "embed"
+
+	"github.com/distribution/reference"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	//go:embed puller.sh
+	pullerSHTextData string
+
+	pullerSHTemplate *template.Template
+)
+
+func init() {
+	t, err := template.New("puller.sh").Parse(pullerSHTextData)
+	if err != nil {
+		panic(err)
+	}
+
+	pullerSHTemplate = t
+}
+
+func renderPullerSH(imgRef string, volDir string) (string, error) {
+	normalizedImgRef, err := reference.ParseDockerRef(imgRef)
```

Comment:
Should we do this check in validation webhook? I was thinking to avoid any err path in this method.

Reply:
Yeah, it makes sense for us to have that check in the webhook. I'll implement that!

But what should we do if we don't have a valid image reference at this point? I know it's the user's fault if they disable the webhook and create an invalid workspace, but I need to do something in code if that function fails. Should I just panic? Or should I continue with the original invalid value? The latter will eventually cause pull/push to fail.

Reply:
Also, the rendering of the Go template can technically fail. But it won't, as long as we have good coverage on that method. So, I can safely make that a panic.

Reply:
I think panic would be the best option for both situations if we don't want to have error handling on that path

Reply:
If cannot normalize, just use the original `imgRef`.
later steps will give users the error.

Reply:
Okay, I've made the changes here. The methods in this package no longer return errors.